### ISSUE TITLE
Improve documentation about what a dAgNode is and how to create one

### ISSUE THREAD

**Issue by satazor** [ipfs/js-ipfs#1754]
This issue is not really a feature request or a bug report, so I will not be using the template.

The [dag.put](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/DAG.md#dagput) method mentions the dagNode argument but it doesn't explain how can we create node with links. Digging into the deprecated object api, it seems we must use https://github.com/ipld/js-ipld-dag-pb. I think that we could improve the document to mention that module and add an example.

**Comment by alanshaw** [comment]
@satazor I agree! Would you have time to send a PR to https://github.com/ipfs/interface-ipfs-core with your findings?

**Comment by satazor** [comment]
Yes I can take a shot on improving it, I will put this on my todo list. I ended up using plain json that worked just fine: https://github.com/ipfs-shipyard/js-ipfs-versidag/blob/initial-implementation/src/index.js, but it wasn’t clear that I could do it.
